recipes/glibmm/all/conanfile.py

```diff
@@ -43,129 +51,115 @@ def _giomm_lib(self):
         return f"giomm-{self._abi_version}"
 
     def validate(self):
-        if hasattr(self, "settings_build") and build.cross_building(self):
+        if hasattr(self, "settings_build") and cross_building(self):
             raise ConanInvalidConfiguration("Cross-building not implemented")
 
         if self.settings.compiler.get_safe("cppstd"):
             if self._abi_version == "2.68":
-                build.check_min_cppstd(self, 17)
+                check_min_cppstd(self, 17)
             else:
-                build.check_min_cppstd(self, 11)
+                check_min_cppstd(self, 11)
 
         if self.options.shared and not self.options["glib"].shared:
             raise ConanInvalidConfiguration(
                 "Linking a shared library against static glib can cause unexpected behaviour."
             )
 
-        if self.options["glib"].shared and microsoft.is_msvc_static_runtime(self):
-            raise ConanInvalidConfiguration(
-                "Linking shared glib with the MSVC static runtime is not supported"
-            )
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+        if self.options["glib"].shared and is_msvc_static_runtime(self):
```

Comment:
```suggestion
        if self.dependencies["glib"].options.shared and is_msvc_static_runtime(self):
```
self.dependencies is better than using old self.options on validate(): https://docs.conan.io/en/latest/reference/conanfile/dependencies.html